spec/storage/s3.spec.js

```diff
@@ -30,63 +33,36 @@ describe("storage/s3", () => {
                 region: null
             }
         };
-        S3 = require("../../lib/storage/s3")(awsSdkMock, promiseMock);
-        s3 = new S3();
-    });
-    describe("configure()", () => {
-        beforeEach(() => {
-            awsSdkMock.S3 = jasmine.createSpy("s3Mock");
-        });
-        it("passes in configuration options for all", () => {
-            expect(awsSdkMock.config.region).toBe("us-east-1");
-            s3.transit();
-            expect(awsSdkMock.S3).toHaveBeenCalledWith({
-                params: {
-                    Bucket: null
-                }
-            });
-        });
-        it("passes in configuration options for all", () => {
-            s3.configure({
-                region: "us-west-1",
-                bucket: "test-bucket"
-            });
-            expect(awsSdkMock.config.region).toBe("us-west-1");
-            s3.transit();
-            expect(awsSdkMock.S3).toHaveBeenCalledWith({
-                params: {
-                    Bucket: "test-bucket"
-                }
-            });
-        });
+        s3 = require("../../lib/storage/s3")(awsSdkMock, {
+            storage: {
+                bucket: "test-bucket",
+                region: "us-east-1"
+            }
+        }, promiseMock);
     });
-    describe("del()", (done) => {
+    describe(".delAsync()", (done) => {
         it("deletes a file", () => {
             s3.delAsync("afile").then((val) => {
                 expect(val).toEqual({
                     Key: "afile"
                 });
+                expect(awsSdkMock.config.region).toBe("us-east-1");
```

Comment:
Exposed transit so we can call it in the tests, but transit isn't used outside of the module.